mne/io/reference.py

```diff
@@ -359,34 +361,49 @@ def set_eeg_reference(inst, ref_channels=None, copy=True, verbose=None):
                          'of Raw, Epochs or Evoked.')
 
     if ref_channels is None:
-        # CAR requested
+        warn('Setting ref_channels=None to compute an average reference is '
+             'deprecated and will be replaced by ref_channels="average" in '
+             '0.16.', DeprecationWarning)
+        ref_channels = 'average'
+
+    if ref_channels != 'average' and projection:
+        raise ValueError('Setting projection=True is only supported for '
+                         'ref_channels="average".')
+
+    if ref_channels == 'average' and projection:  # average reference projector
         if _has_eeg_average_ref_proj(inst.info['projs']):
-            warn('An average reference projection was already added. The data '
-                 'has been left untouched.')
+            warn('An average reference projection was already added. The '
+                 'data has been left untouched.')
             return inst, None
         else:
-            # Creating an average reference may fail. In this case, make sure
-            # that the custom_ref_applied flag is left untouched.
+            # Creating an average reference may fail. In this case, make
+            # sure that the custom_ref_applied flag is left untouched.
             custom_ref_applied = inst.info['custom_ref_applied']
             try:
                 inst.info['custom_ref_applied'] = False
                 inst.add_proj(make_eeg_average_ref_proj(inst.info,
                               activate=False))
-                # If the data has been preloaded, projections will no longer be
-                # automatically applied.
+                # If the data has been preloaded, projections will no
+                # longer be automatically applied.
                 if inst.preload:
-                    logger.info('Average reference projection was added, but '
-                                "hasn't been applied yet. Use the "
-                                '.apply_proj() method function to apply '
-                                'projections.')
-            except:
+                    logger.info('Average reference projection was added, '
+                                'but has not been applied yet. Use the '
+                                'apply_proj method to apply projections.')
+            except:  # TODO: which exception is caught here?
```

Comment:
It's a try/except (but should really be `except Exception`, please change it) to catch any general error and restore the old value of `custom_ref_applied` in the case of failure.